lib/openfeature/sdk/configuration.rb

```diff
@@ -13,13 +13,22 @@ module SDK
     class Configuration
       extend Forwardable
 
-      attr_accessor :context, :provider, :hooks
+      attr_accessor :context, :hooks
+      attr_reader :provider
 
       def_delegator :@provider, :metadata
 
       def initialize
         @hooks = []
       end
+
+      def provider=(provider)
+        @provider.shutdown if @provider.respond_to?(:shutdown)
+
+        provider.init if provider.respond_to?(:init)
```

Comment:
My first (incorrect) read of this thought made me think `provider.init` needed to use the instance variableMight be worth a few comments here, as it took me a minute to realize what is going on:

- shutdown the existing provider if necessary
- startup the new provider if necessary
- finally assign it